### ISSUE TITLE
[3.1.x] Markdown rendering regression

### ISSUE THREAD

**Issue by GawainLynch** [bolt/bolt#5743]
Noticed today by @PhillippOhlandt that bolt.cm wasn't rendering Markdown as HTML, rather just plain text MD source.

Reverting to 3.0.x-dev or earlier fixes problem.

Details
-------

 - Relevant Bolt Version: 3.1.0+ 
 - Install type: any

**Comment by bobdenotter** [comment]
This is prolly related to my changes in how filtering and sanitization is enforced since 3.1.0. Will look into it.

**Comment by bobdenotter** [comment]
I can't really reproduce this.. If I make a simple content type, like this: 

```
dummy:
    name: Dummies
    singular_name: Dummy
    fields:
        title:
            type: text
            class: large
            group: "Block"
        slug:
            type: slug
            uses: [ title ]
        content:
            type: markdown
            height: 150px
```

Both of these will produce proper Markdown output: 
```
    {{ record.content|markdown }}

    {{ fields() }}
```

What happens on bolt.cm, is because we're doing `{{ record|excerpt(600) }}`. As far as I know, 'excerpt' has never parsed markdown, right?

**Comment by PhillippOhlandt** [comment]
@bobdenotter do we really use the `excerpt()` function on the newsitem page? that doesn't make much sense.

**Comment by PhillippOhlandt** [comment]
And, it worked before, so something broke in the codebase.

**Comment by bobdenotter** [comment]
@PhillippOhlandt Yes. we use Excerpt. And I'm trying to thing how "excerpt" could've used correct HTML before. Are you 100% positive it did before? I don't see recent changes in that page: 

https://github.com/bolt/site-v30/commits/master/public/theme/bolt-v300/news.twig

**Comment by PhillippOhlandt** [comment]
Thats the newsitem listing page. But also all newitems were broken and there we shouldn't use `excerpt()`.

**Comment by PhillippOhlandt** [comment]
https://github.com/bolt/site-v30/blob/master/public/theme/bolt-v300/record.twig#L29-L32

**Comment by bobdenotter** [comment]
@PhillippOhlandt Euhm. What?

**Comment by PhillippOhlandt** [comment]
When Gawain updated the site to use the latest 3.1.x, all newsitems showed the raw markdown text.

**Comment by bobdenotter** [comment]
Oooooooh, right. I think i get it now.. currently in 3.1.0, you need to do `{{ record.foo|markdown }}`, but before it used to output markdown when just doing `{{ record.foo }}` if "foo" happened to be a markdown field.

**Comment by PhillippOhlandt** [comment]
Ah, that explained it. So it's more our fault because we went the lazy route and didn't use the proper filter?

**Comment by bobdenotter** [comment]
@rossriley Can you help out, here? I can't find the place where this magic happens anymore.

**Comment by bobdenotter** [comment]
One way to fix this, would be to re-introduce this code to `\Legacy\Content.php`: 

```
                case 'markdown':
                    // Parse the field as Markdown, return HTML
                    $value = $this->app['markdown']->text($this->values[$name]);
                    $value = $this->preParse($value, $allowtwig);
                    $value = new \Twig_Markup($value, 'UTF-8');

                    break;

                case 'html':
```

(see #5611)

But, this is probably not the best way going forward.

**Comment by bobdenotter** [comment]
Note: What happened on bolt.cm was not related to this issue. That was #5601 instead.

**Comment by SahAssar** [comment]
Quick question: Wouldn't this lead to any theme that "adapted to the new way" would double-parse the markdown?

**Comment by bobdenotter** [comment]
@SahAssar Yes, but if they've changed their template this quickly without problems, I'm sure they'll be able to revert that. (I'd think)

**Comment by CarsonF** [comment]
Yeah I'm wondering if this would double parse too...

**Comment by bobdenotter** [comment]
It will, but only if somebody themselves added `{{ record.markdownfield|markdown() }}` to the template. 

There's really no good way to check for that, so they'll just have to revert that. I think the impact will be low though, because it's not like any actual data is lost.
So far, we've bumped into it on bolt.cm, and had no reports from other people about it.

**Comment by CarsonF** [comment]
Is there a way that the markdown filter could check for markdown source and otherwise pass through?

**Comment by bobdenotter** [comment]
Not in a foolproof way. We could "guess" if something is markdown, by checking for HTML tags, but even valid markdown might have embedded tags. So, no.

**Comment by CarsonF** [comment]
Maybe we could wrap the rendered markdown content in a class with a `__toString` method?

**Comment by bobdenotter** [comment]
That's not unlike what Ross suggested we do in the (near) future. For right now we should just fix the bug we introduced in 3.1.0, and be done with it, imho.

**Comment by CarsonF** [comment]
Bokay. I'm not looped in enough to know if would be BC break. 🙊 😶
